cli/azd/pkg/pipeline/pipeline.go

```diff
@@ -175,3 +174,43 @@ var (
 	gitHubYml                string = filepath.Join(gitHubWorkflowsDirectory, defaultPipelineFileName)
 	azdoYml                  string = filepath.Join(azdoPipelinesDirectory, defaultPipelineFileName)
 )
+
+type ciProviderType string
+
+const (
+	ciProviderGitHubActions ciProviderType = "github"
+	ciProviderAzureDevOps   ciProviderType = "azdo"
+)
+
+func toCiProviderType(provider string) (ciProviderType, error) {
+	result := ciProviderType(provider)
+	if result == ciProviderGitHubActions || result == ciProviderAzureDevOps {
+		return result, nil
+	}
+	return "", fmt.Errorf("invalid ci provider type %s", provider)
+}
+
+type infraProviderType string
+
+const (
+	infraProviderBicep     infraProviderType = "bicep"
+	infraProviderTerraform infraProviderType = "terraform"
+	infraProviderUndefined infraProviderType = ""
+)
+
+func toInfraProviderType(provider string) (infraProviderType, error) {
+	result := infraProviderType(provider)
+	if result == infraProviderBicep || result == infraProviderTerraform || result == infraProviderUndefined {
+		return result, nil
+	}
+	return "", fmt.Errorf("invalid infra provider type %s", provider)
+}
+
+type projectProperties struct {
+	CiProvider    ciProviderType
+	InfraProvider infraProviderType
+	RepoRoot      string
+	HasAppHost    bool
+	BranchName    string
+	authType      PipelineAuthType
```

Comment:
Can you make this `AuthType` so it is exported and folks now to set it in the property bag?